src/models/model.cpp

```diff
@@ -277,7 +284,8 @@ DeviceInterface* SetProviderSessionOptions(OrtSessionOptions& session_options,
                                            const std::vector<std::string>& providers,
                                            const std::vector<Config::ProviderOptions>& provider_options_list,
                                            bool is_primary_session_options,
-                                           bool disable_graph_capture) {
+                                           bool disable_graph_capture,
+                                           Config::Model& model_config) {
```

Comment:
```suggestion
                                           const Config::Model& model_config) {
```

Also, shall we pass in the entire Config instead of Config::Model?

Reply:
Done